docs/tutorials/forum-application/8.app.md

```diff
@@ -0,0 +1,422 @@
+			if err != nil {
+				panic(err)
+			}
+			// Append messages to chat history
+			app.onGoingBlock.Set([]byte("history"), []byte(chatHistory))
+			// This adds the user to the DB, but the data is not committed nor persisted until Commit is called
+			respTxs[i] = &abci.ExecTxResult{Code: abci.CodeTypeOK}
+			app.state.Size++
+		}
+	}
+	app.state.Height = req.Height
+
+	response := &abci.FinalizeBlockResponse{TxResults: respTxs, AppHash: app.state.Hash()}
+	return response, nil
+}
+
+// Commit the application state
+func (app *ForumApp) Commit(_ context.Context, _ *abci.CommitRequest) (*abci.CommitResponse, error) {
+	fmt.Println("Executing Application Commit")
+
+	if err := app.onGoingBlock.Commit(); err != nil {
+		panic(err)
+	}
+	saveState(&app.state)
+	return &abci.CommitResponse{}, nil
+}
+
+// ExtendVote returns curse words as vote extensions
+func (app *ForumApp) ExtendVote(_ context.Context, _ *abci.ExtendVoteRequest) (*abci.ExtendVoteResponse, error) {
+	fmt.Println("Executing Application ExtendVote")
+
+	return &abci.ExtendVoteResponse{VoteExtension: []byte(app.CurseWords)}, nil
+}
+
+// VerifyVoteExtension verifies the vote extensions and ensure they include the curse words
+// It will not be called for extensions generated by this validator
+func (app *ForumApp) VerifyVoteExtension(_ context.Context, req *abci.VerifyVoteExtensionRequest) (*abci.VerifyVoteExtensionResponse, error) {
+	fmt.Println("Executing Application VerifyVoteExtension")
+
+	if _, ok := app.valAddrToPubKeyMap[string(req.ValidatorAddress)]; !ok {
+		// we do not have a validator with this address mapped; this should never happen
+		panic(fmt.Errorf("unknown validator"))
+	}
+	curseWords := strings.Split(string(req.VoteExtension), "|")
+	tmpCurseWordMap := make(map[string]struct{})
+	// Verify that we do not have double words and the validator is not trying to cheat us
+	for _, word := range curseWords {
+		tmpCurseWordMap[word] = struct{}{}
+	}
+	if len(tmpCurseWordMap) < len(curseWords) {
+		// Extension repeats words
+		return &abci.VerifyVoteExtensionResponse{Status: abci.VERIFY_VOTE_EXTENSION_STATUS_REJECT}, nil
+	}
+	return &abci.VerifyVoteExtensionResponse{Status: abci.VERIFY_VOTE_EXTENSION_STATUS_ACCEPT}, nil
+}
+
+// State Sync Connection
+
+func (app *ForumApp) ListSnapshots(_ context.Context, _ *abci.ListSnapshotsRequest) (*abci.ListSnapshotsResponse, error) {
+	return &abci.ListSnapshotsResponse{}, nil
+}
+
+func (app *ForumApp) OfferSnapshot(_ context.Context, _ *abci.OfferSnapshotRequest) (*abci.OfferSnapshotResponse, error) {
+	return &abci.OfferSnapshotResponse{}, nil
+}
+
+func (app *ForumApp) LoadSnapshotChunk(_ context.Context, _ *abci.LoadSnapshotChunkRequest) (*abci.LoadSnapshotChunkResponse, error) {
+	return &abci.LoadSnapshotChunkResponse{}, nil
+}
+
+func (app *ForumApp) ApplySnapshotChunk(_ context.Context, _ *abci.ApplySnapshotChunkRequest) (*abci.ApplySnapshotChunkResponse, error) {
+	return &abci.ApplySnapshotChunkResponse{}, nil
+}
```

Comment:
I think you can remove these, as you embedded `abci.BaseApplication` in the app struct